tests/unittests/config/test_cc_apk_configure.py

```diff
@@ -4,102 +4,58 @@
 Test creation of repositories file
 """
 
-import os
 import re
 import textwrap
 
 import pytest
 
-from cloudinit import cloud, helpers, util
+from cloudinit import util
 from cloudinit.config import cc_apk_configure
 from cloudinit.config.schema import (
     SchemaValidationError,
     get_schema,
     validate_cloudconfig_schema,
 )
-from tests.unittests.helpers import (
-    SCHEMA_EMPTY_ERROR,
-    FilesystemMockingTestCase,
-    mock,
-    skipUnlessJsonSchema,
-)
+from tests.unittests.helpers import SCHEMA_EMPTY_ERROR, skipUnlessJsonSchema
+from tests.unittests.util import get_cloud
 
 REPO_FILE = "/etc/apk/repositories"
 DEFAULT_MIRROR_URL = "https://alpine.global.ssl.fastly.net/alpine"
 CC_APK = "cloudinit.config.cc_apk_configure"
 
 
-class TestNoConfig(FilesystemMockingTestCase):
-    def setUp(self):
-        super(TestNoConfig, self).setUp()
-        self.add_patch(CC_APK + "._write_repositories_file", "m_write_repos")
-        self.name = "apk_configure"
-        self.cloud_init = None
-        self.args = []
+class TestApkConfigure:
+    @pytest.fixture(autouse=True)
+    def setup(self, mocker, tmpdir):
+        mocker.patch("cloudinit.temp_utils._ROOT_TMPDIR", str(tmpdir))
+        yield
 
-    def test_no_config(self):
+    @pytest.mark.parametrize(
+        "config",
+        (
+            ({}),
+            ({"apk_repos": {}}),
+            ({"apk_repos": {"alpine_repo": []}}),
+        ),
+    )
+    def test_no_config(self, config, mocker):
```

Comment:
I never get tired of seeing the cosolidation of  pytest via parametrize  and fixutres. thanks! :heart: 

I also appreciate your practice of pulling  the mocker call into the scope of the unittest  method over @mock.patch decorating the method and passing it in as another positional arg. I think it leaves a cleaner method signature as well.